### DISCUSSION TITLE
No response with 'run!(simulation)'

### DISCUSSION THREAD

**Discussion by Zhendong-Hu** [CliMA/Oceananigans.jl#1537] [Computational science!]
using Oceananigans
grid = RegularRectilinearGrid(size=(100, 100, 50), extent=(2π, 2π, 1))
model = IncompressibleModel(grid=grid)
simulation = Simulation(model, Δt=60, stop_time=3600)
run!(simulation)

There isno response with 'run!(simulation)'.

![image](https://user-images.githubusercontent.com/67493671/113381654-16bed900-93b2-11eb-9904-8446b3fea27a.png)

**Comment by ali-ramadhan** [comment]
Hi @Zhendong-Hu does `run!(simulation)` work or is it getting stuck?

You can check the value of `model.clock` which should be at `time = 1 hour, iteration = 60` after you call `run!(simulation)`.

By default `run!(simulation)` produces no output while time-stepping. If you want the simulation to produce output you can pass in a `progress` function to the `Simulation` constructor.

The 2D turbulence example shows how to use a simple `progress` function: https://clima.github.io/OceananigansDocumentation/stable/generated/two_dimensional_turbulence/

For more details and features of the `Simulation` constructor you can check the docstring: https://clima.github.io/OceananigansDocumentation/stable/library/#Oceananigans.Simulations.Simulation-Tuple{Any}

**Reply by glwagner** [reply]
The problem size 100, 100, 50 is huge so that might be the reason its not completing... (we should make the problem size in the README smaller).
